packages/corelib/src/dataModel/Studio.ts

```diff
@@ -84,17 +84,59 @@ export interface DBStudio {
 	routeSets: Record<string, StudioRouteSet>
 	routeSetExclusivityGroups: Record<string, StudioRouteSetExclusivityGroup>
 
+	peripheralDeviceSettings: StudioPeripheralDeviceSettings
+
+	/** Details on the last blueprint used to generate the defaults values for this */
+	lastBlueprintConfig: LastBlueprintConfig | undefined
+}
+
+export interface StudioPeripheralDeviceSettings {
 	/** Contains settings for which Package Containers are present in the studio.
 	 * (These are used by the Package Manager and the Expected Packages)
 	 */
 	packageContainers: Record<string, StudioPackageContainer>
+
 	/** Which package containers is used for media previews in GUI */
 	previewContainerIds: string[]
 	thumbnailContainerIds: string[]
 
-	/** Details on the last blueprint used to generate the defaults values for this */
-	lastBlueprintConfig: LastBlueprintConfig | undefined
+	playoutDevices: ObjectWithOverrides<Record<string, StudioPlayoutDevice>>
+
+	ingestDevices: ObjectWithOverrides<Record<string, StudioIngestDevice>>
+
+	inputDevices: ObjectWithOverrides<Record<string, StudioInputDevice>>
```

Comment:
I'm a little bit unsure about this architecture. One of the advantages of the "generic" Peripheral Device structure, was that it was very easy to just add a new peripheral device and have it working. With this setup, devices need to "neatly" slot into one-and-only-one of these categories, which may turn out to be problematic. Could this be refactored somehow, so that we don't have to categorize the devices so strongly? (Yes, I know that PeripheralDevices have types and subTypes, but that system seems like it could be easily refactored away to something like an "capabilities" array, it's not a core part of the system).

Reply:
Ah, OK, I completely misread this. I can now see, that as far as the previous structure is concerned, these are technically "sub-devices".

Reply:
I've added some comments, which explicitly states them as sub-devices.